dali/device/general.py

```diff
@@ -842,3 +940,524 @@ class DTR2DTR1(_SpecialDeviceCommandTwoParam):
     _addr = 0xc9
     uses_dtr1 = True
     uses_dtr2 = True
+
+
+class UnknownEvent(_DeviceCommand):
+    """A message known to be an "event", but with no specific implementation in this library"""
+
+    @classmethod
+    def from_frame(cls, f, devicetype=0, dev_inst_map=None):
+        return
+
+
+class DeviceInstanceTypeMapper:
+    """
+    Part 103 defines five different event message addressing schemes, as per Table 8 of
+    IEC 62386.103. These are: "Instance", "Device", "Device/Instance", "Device Group" and "Instance
+    Group". Four of these only rely on data contained within the message itself, however the
+    "Device/Instance" addressing scheme does not include information in the 24 bits of the message
+    to indicate the instance type - which means that without further knowledge it is not possible
+    to decode these messages.
+
+    DeviceInstanceTypeMapper stores a mapping from the address and instance number of a device to
+    the instance type, thus allowing a higher-level system to decode "Device/Instance" messages.
+
+    One instance of DeviceInstanceTypeMapper should be created for each DALI bus, and then re-used
+    whenever decoding event messages from that bus through the `from_frame()` method.
+    """
+
+    def __init__(self, initial=None):
+        """
+        Creates a new DeviceInstanceTypeMapper object, optionally with preloaded mappings as
+        defined by `initial`.
+
+        :param initial: A dict of data to preload into the mapping
+        """
+        self._mapping: dict[(int, int), int] = {}
+        if initial:
+            self._mapping = initial
+
+    @property
+    def mapping(self) -> dict[(int, int), int]:
+        return self._mapping
+
+    def autodiscover(
+        self, address_count: int = 63
```

Comment:
Should be 64 to cover addresses 0 to 63 inclusive!

However, it might be nice to support the caller passing in their own iterable of addresses here, and default to range(64) if they don't pass anything in.

Reply:
Yep good idea, I've added this now